src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmRpoPollServiceImpl.java

```diff
@@ -0,0 +1,174 @@
+                    armRpoApi.removeProduction(bearerToken, executionId, userAccount);
+                    log.debug("About to reconcile production files");
+                    armRpoService.reconcileArmRpoCsvData(armRpoExecutionDetailEntity, tempProductionFiles);
+                } else {
+                    log.warn("No production export files found");
+                }
+            } else {
+                log.warn("Create export of production files is still in progress");
+            }
+
+        } catch (Exception e) {
+            log.error("Error while polling ARM RPO", e);
+        } finally {
+            try {
+                cleanUpTempFiles();
+            } catch (Exception e) {
+                log.error("Error while cleaning up ARM RPO polling service temp files", e);
+            }
+
+        }
+    }
+
+    private void cleanUpTempFiles() {
+        for (var tempProductionFile : tempProductionFiles) {
+            if (tempProductionFile.exists()) {
+                if (tempProductionFile.delete()) {
+                    log.info("Deleted temp production file {}", tempProductionFile.getName());
+                } else {
+                    log.warn("Failed to delete temp production file {}", tempProductionFile.getName());
+                }
+            }
+        }
+    }
+
+    private String generateProductionExportFilename(String productionExportFileId) {
+        return "productionExportFileId_" + productionExportFileId + ".csv";
+    }
+
+    private ArmRpoExecutionDetailEntity getArmRpoExecutionDetailEntity(boolean isManualRun) {
+        var armRpoExecutionDetailEntity = armRpoService.getLatestArmRpoExecutionDetailEntity();
+        if (isNull(armRpoExecutionDetailEntity)) {
+            return null;
+        }
+        // If the previous state is saveBackgroundSearch and status is completed
+        // or the previous state is createExportBasedOnSearchResultsTable and status is in progress, return the entity
+        if (saveBackgroundSearchCompleted(armRpoExecutionDetailEntity)
+            || createExportBasedOnSearchResultsTableInProgress(armRpoExecutionDetailEntity)) {
+            return armRpoExecutionDetailEntity;
+        }
+        // If the job is a manual run, the previous status is failed and the state is greater than saveBackgroundSearch, return the entity
+        if (isManualRun && pollServiceNotInProgress(armRpoExecutionDetailEntity)) {
+            return armRpoExecutionDetailEntity;
+        }
+        return null;
+    }
+
+
+    private boolean pollServiceNotInProgress(ArmRpoExecutionDetailEntity armRpoExecutionDetailEntity) {
+        if (nonNull(armRpoExecutionDetailEntity.getArmRpoState())
+            && ArmRpoHelper.failedRpoStatus().getId().equals(armRpoExecutionDetailEntity.getArmRpoStatus().getId())
+            && armRpoExecutionDetailEntity.getArmRpoState().getId() >= ArmRpoHelper.getExtendedSearchesByMatterRpoState().getId()) {
```

Comment:
Is this line testing whether the current state is logically (in terms of API flow) on-or-after the `ExtendedSearchesByMatterRpoState`?

It feels a little bit dangerous to rely on the database ids to determine that. I suppose the existing ones shouldn't change, but they may be added to in the future, so the id ordering does not necessarily correspond to a logical sequence of API calls.

As a safer appropach, consider checking whether the current state is any of an explicit subset.